tests/unit/stats.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <limits>
+
+#include <util/meta.hpp>
+
+/* For randomness, distribution testing we need some
+ * statistical tests... */
+
+namespace testing {
+
+// Simple summary statistics (min, max, mean, variance)
+// computed with on-line algorithm.
+//
+// Note that variance is population variance (i.e. with
+// denominator n, not n-1).
+
+struct summary_stats {
+    int n = 0;
+    double max = -std::numeric_limits<double>::infinity();
+    double min = std::numeric_limits<double>::infinity();
+    double mean = 0;
+    double variance = 0;
+};
+
+template <typename Seq>
+summary_stats summarize(const Seq& xs) {
+    summary_stats s;
+
+    for (auto x: xs) {
+        s.min = x<s.min? x: s.min;
+        s.max = x>s.max? x: s.max;
+        double d1 = x-s.mean;
+        s.mean += d1/++s.n;
```

Comment:
I prefer to not use `++` inside expressions like this. How about `++s.n` as the first line in the loop.

Reply:
Righto.